### ISSUE TITLE
NVDA ignores the aria-label on columnheader in tables

### ISSUE THREAD

**Issue by anna-yeddi** [nvaccess/nvda#19391]
### Brief summary

When `aria-label` is provided to label an empty column header cell or to overwrite the existing string in it, NVDA ignores the explicit accessible name and either announcing the on-screen text of the cell or not announcing anything at all (for empty header cells)

### Steps to reproduce

1. Open `data:text/html,<table><tr><th scope="col" aria-label="First">Visible</th><th scope="col" aria-label="Second hidden"></th></tr><tr><td>foo</td><td>bar</td></tr></table>`
2. With NVDA, navigate between the first and second columns and observe the announcement

### Actual behavior

An NVDA is announcing the following:
1. For the first column with an on-screen string "Visible" and `aria-label="First"`, the "Visible" is announced (`column 1, Visible`)
1. For the second column that is empty and has `aria-label="Second hidden"`, nothing is announced (`column 2`)

### Expected behavior

An NVDA should be announcing the column header's accessible names (that are calculated as "First" and "Second hidden", per Accessibility tree inspection on both Firefox and Chrome):
1. For the first column with an on-screen string "Visible" and `aria-label="First"`, the "First" should be announced (`column 1, First`)
1. For the second column that is empty and has `aria-label="Second hidden"`, the "Second hidden" should be announced (`column 2, Second hidden`)

### NVDA type

installed copy

### NVDA version

2025.3.2.53588

### Windows version

Windows 11 Pro 25H2

### Name and version of other software in use when reproducing the issue

Firefox 146.0 (release) and Chrome 143.0

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by jcsteh** [comment]
Name and content are two different things. Whether the name (aria-label) should override the content is actually not specified anywhere in ARIA. There are cases where it should (e.g. links, buttons) and cases where it shouldn't (e.g. landmarks).

Column headers (and table cells in general) are a tricky case because there can still be semantic content and formatting inside them which would be overridden if the author specified aria-label and we used that instead of the content. We already have this problem for headings, where we do allow the aria-label to override; see #13296. That said, the spec does permit authors to name column headers, so it's reasonable that authors might do it, especially for a column header where the content is normally not large.

I'd suggest that NVDA should at least use the name if there is no content for table cells, like we already do for divs and spans. This can be done by adding more [roles to this check](https://github.com/nvaccess/nvda/blob/ee365d5ae071564b22bf8319f413aecc5fc9ce49/nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp#LL1282). However, I don't know about overriding content where content is present, as that potentially leads to the same problem we have with headings in #13296.

**Comment by anna-yeddi** [comment]
> I'd suggest that NVDA should at least use the name if there is no content for table cells, like we already do for divs and spans.

Sounds great to me. Thank you!

**Comment by seanbudd** [comment]
>I'd suggest that NVDA should at least use the name if there is no content for table cells, like we already do for divs and spans. This can be done by adding more [roles to this check](https://github.com/nvaccess/nvda/blob/ee365d5ae071564b22bf8319f413aecc5fc9ce49/nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp#LL1282). However, I don't know about overriding content where content is present, as that potentially leads to the same problem we have with headings in https://github.com/nvaccess/nvda/issues/13296.

We agree with this and would accept a PR to fix this for empty cells
